### PR TITLE
Allow overriding relationship ids

### PR DESCRIPTION
**Pull request by lchanmann**
## Changes
- Allow overriding relationship ids
- Override user capability junction ids

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/dao/Relationship.js [hunk 1]
```diff
@@ -182,6 +182,14 @@ foam.CLASS({
         Only used for many to many relationships.
         Name of the class that this junction class should inherit from.
       `
+    },
+    {
+      class: 'StringArray',
+      name: 'ids',
+      documentation: "Ids of the relationship.",
+      factory: function() {
```
Comment (kgrgreer) [inline src/foam/dao/Relationship.js:190]: Doesn't just using value: work?

File: src/foam/dao/Relationship.js [hunk 2]
```diff
@@ -300,7 +308,7 @@ foam.CLASS({
         package: this.package,
         name: name,
         extends: this.extends,
-        ids: ['sourceId', 'targetId'],
+        ids: this.ids,
```
Comment (kgrgreer) [inline src/foam/dao/Relationship.js:311]: How is this different?

File: src/foam/nanos/crunch/Capability.js [hunk 3]
```diff
@@ -449,6 +449,7 @@ foam.RELATIONSHIP({
   extends:'foam.nanos.crunch.Renewable',
   sourceModel: 'foam.nanos.auth.User',
   targetModel: 'foam.nanos.crunch.Capability',
+  ids: [ 'id' ],
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/Capability.js:452]: I think this is the only important line.

File: src/foam/dao/Relationship.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/dao/Relationship.js:190]: value works.

File: src/foam/dao/Relationship.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/dao/Relationship.js:311]: from hardcoded to property.

File: src/foam/nanos/crunch/Capability.js (hunk 3 above)
Comment (lchanmann) [reply src/foam/nanos/crunch/Capability.js:452]: yes, that's when relationship allows setting 'ids' to a custom one. The default ids is ['sourceId', 'targetId'].

File: src/foam/dao/Relationship.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/dao/Relationship.js:190]: Well, value doesn't work. StringArray property has a default factory and factory is evaluated before value.

**Pull request merged**